File: src/com/intellij/advancedExpressionFolding/expression/Asin.java
package com.intellij.advancedExpressionFolding.expression;

import com.intellij.openapi.util.TextRange;
import com.intellij.psi.PsiElement;

import java.util.List;

public class Asin extends Function implements ArithmeticExpression {
    public Asin(PsiElement element, TextRange textRange, List<Expression> operands) {
        super(element, textRange, "asin", operands);
    }
}

File: src/com/intellij/advancedExpressionFolding/expression/math/trig/Asin.kt
package com.intellij.advancedExpressionFolding.expression.math.trig

import com.intellij.advancedExpressionFolding.expression.Expression
import com.intellij.advancedExpressionFolding.expression.Function
import com.intellij.advancedExpressionFolding.expression.math.ArithmeticExpression
import com.intellij.openapi.util.TextRange
import com.intellij.psi.PsiElement
import java.util.List

class Asin(
    element: PsiElement,
    textRange: TextRange,
    operands: List<Expression>
) : Function(element, textRange, "asin", operands), ArithmeticExpression

